File: tests/fixtures/sample.ts
/**
 * Sample TypeScript file for testing parser
 */

// A simple function
export function add(a: number, b: number): number {
  return a + b;
}

// A class with methods
export class Calculator {
  private history: number[] = [];

  /**
   * Multiply two numbers
   */
  public multiply(a: number, b: number): number {
    const result = a * b;
    this.history.push(result);
    return result;
  }

  /**
   * Get calculation history
   */
  public getHistory(): number[] {
    return this.history;
  }
}

// An interface
export interface User {
  id: string;
  name: string;
  email: string;
}

// A type alias
export type UserId = string | number;

// An enum
export enum Status {
  Active = 'active',
  Inactive = 'inactive',
  Pending = 'pending',
}

// An async function
export async function fetchData(url: string): Promise<any> {
  const response = await fetch(url);
  return response.json();
}

File: tests/fixtures/sample.py
"""
Sample Python file for testing parser
"""

def add(a: int, b: int) -> int:
    """Add two numbers together."""
    return a + b


class Calculator:
    """A simple calculator class."""

    def __init__(self):
        self.history = []

    def multiply(self, a: int, b: int) -> int:
        """Multiply two numbers."""
        result = a * b
        self.history.append(result)
        return result

    def get_history(self) -> list:
        """Get calculation history."""
        return self.history


@dataclass
class User:
    """User data class."""
    id: str
    name: str
    email: str


async def fetch_data(url: str) -> dict:
    """Fetch data from URL asynchronously."""
    async with aiohttp.ClientSession() as session:
        async with session.get(url) as response:
            return await response.json()
